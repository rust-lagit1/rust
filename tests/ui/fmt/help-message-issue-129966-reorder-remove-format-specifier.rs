// Test checks for the help messages in the error output/
//@ check-run-results

struct Foo(u8, u8);

fn main() {
    let f = Foo(1, 2);
    format!("{f:?#}");
//~^ ERROR invalid format string: unknown format identifier '?#'
    format!("{f:?x}");
//~^ ERROR invalid format string: unknown format identifier '?x'
    format!("{f:?X}");
//~^ ERROR invalid format string: unknown format identifier '?X'
}
