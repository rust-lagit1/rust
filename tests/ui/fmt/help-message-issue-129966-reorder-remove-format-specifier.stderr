error: invalid format string: unknown format identifier '?#'
  --> $DIR/help-message-issue-129966-reorder-remove-format-specifier.rs:8:17
   |
LL |     format!("{f:?#}");
   |                 ^^ help: the format parameter should be written `#?` in format string
   |
   = note: if you intended to print `{`, you can escape it using `{{`

error: invalid format string: unknown format identifier '?x'
  --> $DIR/help-message-issue-129966-reorder-remove-format-specifier.rs:10:18
   |
LL |     format!("{f:?x}");
   |                  ^ help: consider removing `x` in format string
   |
   = note: if you intended to print `{`, you can escape it using `{{`

error: invalid format string: unknown format identifier '?X'
  --> $DIR/help-message-issue-129966-reorder-remove-format-specifier.rs:12:18
   |
LL |     format!("{f:?X}");
   |                  ^ help: consider removing `X` in format string
   |
   = note: if you intended to print `{`, you can escape it using `{{`

error: aborting due to 3 previous errors

